astroquery/mast/tests/test_mast.py

```diff
@@ -91,3 +113,104 @@ def test_observations_query_object_async(patch_post):
 def test_observations_query_object(patch_post):
     result = mast.Observations.query_object("M103", radius=".02 deg")
     assert isinstance(result, Table)
+
+
+def test_query_criteria_async(patch_post):
+    # without position
+    responses = mast.Observations.query_criteria_async(dataproduct_type=["image"],
+                                                       proposal_pi="Ost*",
+                                                       s_dec=[43.5, 45.5])
+    assert isinstance(responses, list)
+
+    # with position
+    responses = mast.Observations.query_criteria_async(filters=["NUV", "FUV"],
+                                                       objectname="M101")
+    assert isinstance(responses, list)
+
+
+def test_query_criteria(patch_post):
+    # without position
+    result = mast.Observations.query_criteria(dataproduct_type=["image"],
+                                              proposal_pi="Ost*",
+                                              s_dec=[43.5, 45.5])
+    assert isinstance(result, Table)
+
+    # with position
+    result = mast.Observations.query_criteria(filters=["NUV", "FUV"],
+                                              objectname="M101")
+    assert isinstance(result, Table)
+
+
+# count functions
+def test_observations_query_region_count(patch_post):
+    result = mast.Observations.query_region_count(regionCoords, radius="0.2 deg")
+    assert isinstance(result, np.int64)
```

Comment:
Great will do! Was trying to figure out what was going wrong with these tests. :)